[contributing] New doc on Gatsby's Docs Writing Process (#21945)
gatsbyjs/gatsby: docs/contributing/docs-writing-process.md

The knowledge of how to work with a technique, source plugin, or varied use case may be known internally to Gatsby team members, but it also may only exist on the web at large. This contributing doc is intended to provide a written process for producing docs without prior information, a critical function of the Gatsby Learning team and open source community.

## Identifying a topic

When identifying a topic, start by:

1. Looking at GitHub issues. Good labels to seek out are [`help wanted`](https://github.com/gatsbyjs/gatsby/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22+) and [`good first issue`](https://github.com/gatsbyjs/gatsby/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).

   - This part may be covered if a docs issue is assigned to you, or if you’re signing up voluntarily to take on an issue.

2. Look at the [stub list](/contributing/stub-list/) to find docs that need contributions.

3. Look at the [learning workflow meta issue](https://github.com/gatsbyjs/gatsby/issues/13708) to find active areas looking for docs.

> 4. Read through the existing Gatsby docs information and find gaps in topic coverage. Is there an area you feel is missing? [File an issue](/contributing/how-to-file-an-issue/) first to discuss, then implement in a PR.

Author: Issue feedback sounds good! I've seen quite a few that haven't waited for approval, but we can always reiterate that recommendation.

5. Observe common points of confusion or rough edges through user feedback and recommend solutions.

> _Note: It’s required to open a GitHub issue before submitting a PR if one does not already exist._

## Selecting the correct format

> A GitHub issue for new learning material should indicate the format. Is it a Reference or Conceptual Guide? A Tutorial? A recipe?

Author: I could add some links. But ever since your comment about branching while reading, I've limited how many links I put inline. So they'd likely come after the section.

> A GitHub issue for new learning material should indicate the format. Is it a Reference or Conceptual Guide? A Tutorial? A recipe?

Reviewer: Didn't realize I had such an effect! In my mind there is actually a bit of a difference here. I consider this glossary type links. If people need more context, they'll click. Versus links we expect them to click on and review in addition to what they're currently reading.

Does docs coverage exist anywhere on gatsbyjs.org? If so, would an alternative format help provide information for Gatsby learners of different skill and experience levels? For example, if a tutorial exists but there is no coverage in Reference Guides, adding more content in a different format would benefit users.

- Tutorials vs. recipes vs. guides
  - **Tutorials** are step-by-step learning materials that show a user how to do something without skipping steps. Tutorials must consider a beginner experience, but can be useful for anyone.
  - **Recipes** are shorter, more concise tutorials that limit their scope to a single task or outcome by using prerequisites and limited steps. Additional resources at the end lead the user to a logical next step. Recipes are for developers of all skill levels.

>   - **Guides** are different than tutorials, and the two main sections are Reference Guides and Conceptual Guides. Instead of “steps 1-4”, guides use standalone headings and sections to explain how to accomplish the task. The audience is often more advanced than beginner due to the subject matter, but should be approachable by anyone.

Reviewer: ```suggestion
  - **Guides** are different than tutorials, and the two main sections are Reference Guides and Conceptual Guides. Instead of “steps 1-4”, guides use standalone headings and sections to explain how to accomplish the task. They are often aimed at a more advanced audience due to the subject matter but should be approachable to anyone.
```

> Given the difference in audiences of the 3 main learning material formats, that’s where increasing and overlapping coverage of topics can help support Gatsby learners of different skill and experience levels.

Reviewer: ```suggestion
Given the difference in audiences of the three main learning material formats, increasing and overlapping coverage of topics can help support Gatsby learners of different skill and experience levels.
```
Do we spell out numbers under ten?

Author: We don't have a guideline on it. I typically avoid it when a number leads a sentence or headline, but am fine with them in the text.

> Unless for a specific purpose that follows the [Gatsby blog post guidelines](/contributing/blog-contributions/), do not suggest blog posts when what is needed is user documentation.

Reviewer: ```suggestion
Please follow the [Gatsby blog post guidelines](/contributing/blog-contributions/) and do not suggest blog posts when what is needed is user documentation.
```

More on the format of documentation and learning materials can be found in the contributing docs:

- [Docs Templates](/contributing/docs-templates/)
- [Gatsby Style Guide](/contributing/gatsby-style-guide/)

## Gathering and validating supporting information

> Writing an effective doc that meets the needs of Gatsby users requires gathering information from various sources and applying those concepts to your original writing. You must digest and validate the details you uncover and understand them enough to describe in a way that users of multiple skill levels can learn.

Reviewer: ```suggestion
Writing an effective doc that meets the needs of Gatsby users requires gathering information from various sources and applying those concepts to your original writing. You must digest and validate the details you uncover and understand them enough to describe them in a way that users of multiple skill levels can learn.
```

3. You can also search the web for additional examples outside of the Gatsby GitHub org or docs site. Be sure to check Gatsby versions and only reference the most current examples unless for a specific purpose.

4. In the event there is no information available to write a greenfield doc after trying all the above steps, such as for a new integration, try interviewing Gatsby core team members to help to produce an outline and content tips. See [Pair Programming program](/contributing/pair-programming/).

5. **For Gatsby inkteam members:** You can also search in the Gatsby Slack internally for related information. It’s possible other team members have discussed the issue, or even compiled trouble points into a doc. Look and ask around as part of your discovery work.

After you’ve collected supporting information, you must produce original writing to be accepted in the Gatsby docs. Copying other blog posts, materials, or Gatsby team member interviews word-for-word without attribution is not acceptable or allowed. Furthermore, direct quotes from interviews are also _almost never_ effective for guides, recipes, or tutorials.

> Rather, the best way to write greenfield docs is with new text that explains the concepts and ideas you uncovered. With research and demos supporting your writing, can speak more legitimately to use cases and common troubleshooting. Phrasing information in alternative ways for different skill levels is a bit like avoiding usage of the same word or phrase in its definition: if a reader didn’t understand it the first time, they may be more likely to understand it with an alternative explanation.

Reviewer: ```suggestion
Rather, the best way to write greenfield docs is with new text that explains the concepts and ideas you uncovered. With research and demos supporting your writing, you can speak more legitimately to use cases and common troubleshooting. Phrasing information in alternative ways for different skill levels is a bit like avoiding usage of the same word or phrase in its definition: if a reader didn’t understand it the first time, they may be more likely to understand it with an alternative explanation.
```

## Producing learning materials

> Create demo sites and examples to provide more authoritative material that supports developer experience. When relevant, **functioning source code is a requirement to writing docs** that truly educate users. Source code examples can also be linked from a doc as an additional resource. Include testing as part of your source code to ensure it is robust and stands the test of time.

Reviewer: ```suggestion
Create demo sites and examples to provide more authoritative material that supports the developer experience. When relevant, **functioning source code is a requirement to writing docs** that truly educate users. Source code examples can also be linked from a doc as an additional resource. Include testing as part of your source code to ensure it is robust and stands the test of time.
```

Follow the [docs templates](/contributing/docs-templates/) to ensure you’re producing content in the right format for its purpose.

Use the [Markdown syntax doc](/docs/mdx/markdown-syntax/) to understand your options for formatting text with Markdown, and follow [accessibility recommendations](/docs/making-your-site-accessible/#how-to-improve-accessibility) for [heading levels](/contributing/docs-contributions/#headings) and image alt text.

Run the docs site locally to check formatting and functionality. There are instructions in the [contributing docs](/contributing/docs-contributions/).

> Produce images and rich media following a consistent specification, which will be published on this site soon.

Reviewer: Should we add this when we don't have that spec yet?

Author: I debated on that too.... probably not, so I'll remove it.

Refer to the [Gatsby style guide](/contributing/gatsby-style-guide/) to ensure your PR will be accepted.

## Submitting a pull request

> When submitting work to fulfill an issue, team members are allowed to request feedback in a Google Doc for comments inline. However, this format does not count toward a story deadline internally: only a PR does.

Reviewer: Should this be part of an internal doc versus an external one? Or if it should be here, perhaps it can go after the information in this section that applies to all contributors?

Author: I debated on whether to include this here at all; I think it could go in the Learning handbook instead. Whether anyone actually reads it is another story 🤷‍♀
